cloudtasks/pull_queue/main.go

```diff
@@ -0,0 +1,194 @@
+	}
+}
+
+// [START cloud_tasks_create_task]
+// taskCreate creates a new Task on the specified pull queue.
+// TODO(developer) call this with your Project, and Location & Queue IDs of your pull queue.
+func taskCreate(projectID, locationID, queueID string) (*taskspb.Task, error) {
+	// Create a new Cloud Tasks client instance.
+	// See https://godoc.org/cloud.google.com/go/cloudtasks/apiv2beta2
+	ctx := context.Background()
+	c, clientErr := cloudtasks.NewClient(ctx)
+	if clientErr != nil {
+		return nil, clientErr
+	}
+
+	// Message to be sent as the task payload.
+	message := base64.StdEncoding.EncodeToString([]byte("a message for the recipient"))
+
+	// Construct the expected form of the Queue ID.
+	queueName := fmt.Sprintf("projects/%s/locations/%s/queues/%s", projectID, locationID, queueID)
+
+	// Cloud Tasks Go Client uses protobuf.
+	// See https://godoc.org/google.golang.org/genproto/googleapis/cloud/tasks/v2beta2#CreateTaskRequest
+	req := &taskspb.CreateTaskRequest{
+		Parent: queueName,
+		Task: &taskspb.Task{
+			PayloadType: &taskspb.Task_PullMessage{
+				PullMessage: &taskspb.PullMessage{
+					Payload: []byte(message),
+				},
+			},
+		},
+	}
+
+	createdTask, reqErr := c.CreateTask(ctx, req)
+	if reqErr != nil {
+		return nil, reqErr
+	}
+
+	fmt.Println("Created task:", createdTask.GetName())
+
+	return createdTask, nil
+}
+// [END cloud_tasks_create_task]
+
+// [START cloud_tasks_lease_and_acknowledge_task]
+// runTaskPull leases the next task from the specified pull queue.
+// TODO(developer) call this with your Project, and Location & Queue IDs of your pull queue.
+func taskLease(projectID, locationID, queueID string) (*taskspb.Task, error) {
+	// Create a new Cloud Tasks client instance.
+	// See https://godoc.org/cloud.google.com/go/cloudtasks/apiv2beta2
+	ctx := context.Background()
+	c, clientErr := cloudtasks.NewClient(ctx)
+	if clientErr != nil {
+		return nil, clientErr
+	}
+
+	// Construct the expected form of the Queue ID.
+	queueName := fmt.Sprintf("projects/%s/locations/%s/queues/%s", projectID, locationID, queueID)
+
+	// Cloud Tasks Go Client uses protobuf.
+	// See https://godoc.org/google.golang.org/genproto/googleapis/cloud/tasks/v2beta2#LeaseTasksRequest
+	req := &taskspb.LeaseTasksRequest{
+		Parent:        queueName,
+		MaxTasks:      1,
+		LeaseDuration: &duration.Duration{Seconds: 600},
+		ResponseView:  taskspb.Task_FULL,
+	}
+
+	// See https://godoc.org/google.golang.org/genproto/googleapis/cloud/tasks/v2beta2#LeaseTasksResponse
+	resp, reqErr := c.LeaseTasks(ctx, req)
+	if reqErr != nil {
+		return nil, reqErr
+	}
+
+	// If no tasks are available, nothing further to be done.
+	if len(resp.Tasks) == 0 {
+		return nil, nil
+	}
+
+	// Leasing tasks allows retrieval of one or more tasks. The Tasks property will
+	leasedTask := resp.Tasks[0]
+
+	// See the full code on Github for the implementation of toJsonString.
+	fmt.Println("Leased task:", toJSONString(leasedTask))
```

Comment:
I'd prefer to print it directly or some particular property than use a helper to format. Maybe there is a canonical code sample for this?

Reply:
In this case I was matching the Nodejs and Python output fairly closely. I can skip the JSON formatting and more closely match the Java sample instead.